### PR TITLE
[Block-STM] Minor improvements to the block-stm benchmark

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

- Add support for metric publisher. 
- Add support for running stand-alone block-stm parallel execution. 

copilot:summary

### CODE REVIEW

**Comment by gelash** [comment]
Thanks Sital! LGTM
Adding @danielxiangzl as well to be up-to-date on the benchmarks.

**Review by gelash — APPROVED** [review]

File: aptos-move/aptos-transaction-benchmarks/src/main.rs [hunk 1]
```diff
@@ -2,44 +2,88 @@
...
+    pub num_accounts: Vec<usize>,
+
+    #[clap(long)]
+    pub num_txns: Option<Vec<usize>>,
+
+    #[clap(long, default_value = "true")]
+    pub run_parallel: bool,
+
+    #[clap(long, default_value = "true")]
+    pub run_sequential: bool,
```
Comment (msmouse) [inline aptos-move/aptos-transaction-benchmarks/src/main.rs:44]: Can you actually pass `--run-sequential=false`? (I don't think so)

File: aptos-move/aptos-transaction-benchmarks/src/main.rs [hunk 2]
```diff
@@ -2,44 +2,88 @@
...
+    ParallelExecution(ParallelExecutionOpt)
+}
+
+#[derive(Debug, Parser)]
+struct ParallelAndSeqOpt {
+    #[clap(long, default_value = "10000")]
+    pub num_accounts: Vec<usize>,
+
+    #[clap(long)]
+    pub num_txns: Option<Vec<usize>>,
```
Comment (msmouse) [inline aptos-move/aptos-transaction-benchmarks/src/main.rs:38]: I'd call it `block_sizes`

File: aptos-move/aptos-transaction-benchmarks/src/main.rs [hunk 3]
```diff
@@ -70,20 +114,62 @@ fn main() {
...
+
+}
+
+fn parallel_execution(opt: ParallelExecutionOpt) {
+    aptos_logger::Logger::new().init();
+    START_TIME.set(
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_millis() as i64,
```
Comment (msmouse) [inline aptos-move/aptos-transaction-benchmarks/src/main.rs:142]: use `duration_since_epoch()`?

**Comment by sitalkedia** [comment]
Thanks for the review folks, I am actually bundling this change with a much broader block-stm sharding change I am working on. Will send our a PR for that next week. Closing this at the moment.

**Comment by danielxiangzl** [comment]
Thanks for improving the BlockSTM benchmark! 
Just a heads up, you may also need to change the command inside execution benchmark to be consistent with your changes:
https://github.com/aptos-labs/aptos-core/blob/main/testsuite/sequential_execution_performance.py#L13
https://github.com/aptos-labs/aptos-core/blob/main/testsuite/parallel_execution_performance.py#L47

**Pull request closed without merging**